### ISSUE TITLE
Make "no-useless-constructor" rule fixable

### ISSUE THREAD

**Issue by OliverSieweke** [eslint/eslint#11693]
**What rule do you want to change?**

[`no-useless-constructor`](https://eslint.org/docs/rules/no-useless-constructor)

**Does this change cause the rule to produce more or fewer warnings?**

Exactly the same warnings.

**How will the change be implemented? (New option, new default behavior, etc.)?**

The `--fix` option will fix some of the problems reported by this rule.

**Please provide some example code that this change will affect:**

```js
class A {
    constructor () {}
}
```
```js
class A extends B {
    constructor (...args) {
      super(...args);
    }
}
```

**What does the rule currently do for this code?**

Currently the rule merely reports a problem.

**What will the rule do after it's changed?**

After the rule is changed the `--fix` option will transform the above code snippets to the following:
```js
class A {}
```
```js
class A extends B {}
```

**Are you willing to submit a pull request to implement this change?**

Yes

**Comments**

There are some restricted cases that are reported by this rule for which it would not be completely safe to remove the constructor - precisely those cases in which not **all** arguments of the constructor are passed down to `super()`. 
Compare the following code snippets for example:

```js
class B {
    constructor(...args) {
        console.log(...args);
    }
}

class A extends B {
    constructor () {                    // Error: Useless constructor.
        super();                        // Not passing down any arguments
    }
}

class A2 extends B {
    constructor(value) {                // Error: Useless constructor.
        super(value);                   // Only passing down one argument
    }
}

const a = new A("test1", "test2");      // Nothing logged
const a2 = new A2("test1", "test2");    // "test1" logged
```
Removing the "useless" constructor:
```js
class B {
    constructor(...args) {
        console.log(...args);
    }
}

class A extends B {}

const a = new A("test1", "test2");      // "test1 test2" logged
```

It may therefore be preferable to only automatically fix the cases in which all the arguments of the constructor are passed down to `super()`.
